src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(
+      eventName,
+      () => {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      },
+      { once: true },
+    );
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  document.body.insertAdjacentHTML(
+    'beforeend',
+    `<div class="message">${message}</div>`,
+  );
```

Comment:
you need to replace existing div:
![image](https://github.com/mate-academy/js_promise_function_DOM/assets/102289654/e15d4177-ae28-4323-ac5c-bf583898a329)